### PR TITLE
VSO => Codespaces

### PR DESCRIPTION
**Pull request by edgonmsft**
Fixing the symlinks in universal image.

### CODE REVIEW

**Comment by Chuxel** [comment]
Doing a quick local build to verify. Just to check, do you need this released ASAP?

**Comment by edgonmsft** [comment]
> Doing a quick local build to verify. Just to check, do you need this released ASAP? (Not only to the "dev" tag, but 0.5.0)

I was intending to do a different PR once I change the user to codespace, but was probably going to do both at the same time. Still checking if there is something especial I need to fix,

**Comment by Chuxel** [comment]
Ok, this one built fine - do you want me to hold on merging for the codespace update or just merge this one?

**Comment by Chuxel** [comment]
Main thing I don't know is whether @chrmarti will do a release tomorrow for another reason.  You could also just leave the version number alone here and bump it with the codespaces change.

**Comment by edgonmsft** [comment]
Lets hold of, by tomorrow I'll know how hard its going to be.

**Comment by Chuxel** [comment]
Cool - added the WIP label so we don't do it by mistake.  Feel free to ping for another review when you are ready.

**Comment by edgonmsft** [comment]
@Chuxel I made the changes necesary. There are still 3 things that say vsonline because they require other changes:
- I didn't change the folder name from vsonline-linux.
- A folder inside the container is still called .vsonline since that requires changes on our side.
- An environment variable is used to track by the oryx team.

**Comment by Chuxel** [comment]
You can rename the folder as well - there's one spot that needs tweaking here is the only related change: https://github.com/microsoft/vscode-dev-containers/blob/bd709c268bbc3f5649431509bcf75998fde332b4/build/config.json#L16

I'll run a build locally to check it out.

**Comment by Chuxel** [comment]
Yeah that needs to match the folder name -- so the folder can now be renamed to codespaces-linux. :)

**Comment by edgonmsft** [comment]
Just pushed that change too.

**Comment by Chuxel** [comment]
Built succesfully. The services is "codspaces" with an s right?  Should the folder have that name (codespaces-linux)?

Also the README and devcontainer.json name still say VS Online... I assume that should be VS Codespaces?

**Comment by edgonmsft** [comment]
I had missed the instances of VS Online and Visual Studio Online

File: containers/codespaces-linux/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -26,9 +26,9 @@ ENV SHELL=/bin/bash
 ENV ORYX_ENV_TYPE=vsonline-present
```
Comment (rajkumar42) [inline containers/codespaces-linux/.devcontainer/Dockerfile:26]: We can change this too, but let Oryx team know.

**Comment by Chuxel** [comment]
Looping in Nik just to validate naming.  Currently it's:

- Codespaces for the definition (not VS Codespaces)
- codespace (singular) for the user

@nikmd23  is that in line with guidance?

File: containers/codespaces-linux/.devcontainer/Dockerfile [hunk 2]
```diff
@@ -85,39 +85,39 @@ RUN apt-get update -yq \
...
-    && usermod -a -G docker vsonline \
+    && usermod -a -G docker codespace \
     #
     # Setup .NET Core
     # Hack to get dotnet core sdks in the right place - Oryx images do not put dotnet on the path because it will break AppService.
-    # The following script will put the dotnet's at /home/vsonline/.dotnet folder where dotnet will look by default.
-    && mv /tmp/vsonline/symlinkDotNetCore.sh /home/vsonline/symlinkDotNetCore.sh \
-    && sudo -u vsonline /bin/bash /home/vsonline/symlinkDotNetCore.sh 2>&1 \
-    && rm /home/vsonline/symlinkDotNetCore.sh \
+    # The following script will put the dotnet's at /home/codespace/.dotnet folder where dotnet will look by default.
```
Comment (rajkumar42) [inline containers/codespaces-linux/.devcontainer/Dockerfile:103]: Have we made corresponding changes in Cascade repo?

File: containers/codespaces-linux/.devcontainer/devcontainer.json [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 	"dockerFile": "Dockerfile",
-	"remoteUser": "vsonline",
-	"workspaceMount": "source=${localWorkspaceFolder},target=/home/vsonline/workspace,type=bind,consistency=cached",
-	"workspaceFolder": "/home/vsonline/workspace",
+	"remoteUser": "codespace",
+	"workspaceMount": "source=${localWorkspaceFolder},target=/home/codespace/workspace,type=bind,consistency=cached",
+	"workspaceFolder": "/home/codespace/workspace",
 	"runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined" ],
 
 	// Add the IDs of extensions you want installed when the container is created.
```
Comment (rajkumar42) [inline containers/codespaces-linux/.devcontainer/devcontainer.json:9]: Is the extension  ms-vsonline.vsonline going to be renamed?

**Comment by Chuxel** [comment]
@edgonmsft Ok checked with Nik and JC 

1. The README should say "Visual Studio Codespaces" since it's not constrained.
2. Since the "name" in devcontainer.json shows up in the lower left, that could be left Codespaces.
3. codespace (singluar) is fine for the user name.

File: containers/codespaces-linux/.devcontainer/Dockerfile (hunk 1 above)
Comment (edgonmsft) [reply containers/codespaces-linux/.devcontainer/Dockerfile:26]: Yep thats why I haven't changed it yet.

File: containers/codespaces-linux/.devcontainer/Dockerfile (hunk 2 above)
Comment (edgonmsft) [reply containers/codespaces-linux/.devcontainer/Dockerfile:103]: Yep the first part is already merged in.

File: containers/codespaces-linux/.devcontainer/devcontainer.json (hunk 3 above)
Comment (edgonmsft) [reply containers/codespaces-linux/.devcontainer/devcontainer.json:9]: Yes but not yet, so its going to be a different change and needs a little coordination.

**Comment by edgonmsft** [comment]
Updated the readme with the renames, do they look good to you @Chuxel ?

**Comment by Chuxel** [comment]
LGTM!  Do you want me to merge?

**Comment by edgonmsft** [comment]
Yep thanks!

**Pull request merged**